app/models/miq_provision_virt_workflow.rb

```diff
@@ -1092,4 +1111,17 @@ def selected_host(src)
     raise _("Unable to find Host with Id: [%{id}]") % {:id => src[:host_id]} if src[:host].nil?
     [load_ar_obj(src[:host])]
   end
+
+  def log_allowed_template_list(template_list)
+    if template_list.blank?
```

Comment:
If this is an active record object, it causes a query. The `length` may cause another query

This seems like a debug only thing

BUT this is the original code, so leave it be